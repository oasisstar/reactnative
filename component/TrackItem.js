import React, { Component } from 'react';
import { StyleSheet, View, ImageBackground, Text, TouchableOpacity} from 'react-native';

class TrackItem extends Component {
    constructor(props){
        super(props)
    }

    render() {
        return (
            <View style={styles.detailView}>
                <Text style={styles.valueText}>{this.props.text}</Text>
                <View style={styles.valueView}>
                    <Text style={styles.valueText}>{this.props.value}</Text>
                </View>
            </View>
        );
    }
}

const styles = StyleSheet.create({

    detailView: {
        marginTop: 20,
		alignSelf: "stretch",
		alignItems: 'center',
	},

	valueText: {
        fontSize: 25,
        color: '#5D81A3',
        textAlign: "center"
	},
	
	valueView: {
        marginTop: 10,
		borderRadius: 15,
		borderColor: '#5D81A3',
        borderWidth: 2,
		alignItems: 'center',
		justifyContent: 'center',
		alignSelf: "stretch",
	}
});
export default TrackItem;