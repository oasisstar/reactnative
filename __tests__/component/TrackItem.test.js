// __tests__/TrackItem-test.js
import React from 'react';
import TrackItem from '../../component/TrackItem';

import renderer from 'react-test-renderer';

test('renders correctly', () => {
  const tree = renderer.create(<TrackItem text="TestName" value="TestValue"/>).toJSON();
  expect(tree).toMatchSnapshot();
});