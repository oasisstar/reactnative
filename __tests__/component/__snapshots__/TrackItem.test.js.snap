// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "alignSelf": "stretch",
      "marginTop": 20,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#5D81A3",
        "fontSize": 25,
        "textAlign": "center",
      }
    }
  >
    TestName
  </Text>
  <View
    style={
      Object {
        "alignItems": "center",
        "alignSelf": "stretch",
        "borderColor": "#5D81A3",
        "borderRadius": 15,
        "borderWidth": 2,
        "justifyContent": "center",
        "marginTop": 10,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#5D81A3",
          "fontSize": 25,
          "textAlign": "center",
        }
      }
    >
      TestValue
    </Text>
  </View>
</View>
`;
